### ISSUE TITLE
XML-resolver.jar and jing.jar cannot be located

### ISSUE THREAD

**Issue by MonikaBarget** [BaseXdb/basex#1872]
I would like to use the BaseX GUI but am having issues with the installation. I did not get an error notification during the actual installation process, but when I call "basex -h" or "basexgui -h", Ubuntu tells me that Java components are missing. In the beginning, XML-resolver.jar, jing.jar and tagsoup.jar could not be located in console mode. Then I tried updating / re-installing Java altogether, and now tagsoup.jar is found, but the other two are still not located:

```
mbwiesbaden@ubuntu-wiesbaden:~$ basex -h
[warning] /usr/bin/basex: Unable to locate /usr/share/java/xml-resolver.jar in /usr/share/java
[warning] /usr/bin/basex: Unable to locate /usr/share/java/jing.jar in /usr/share/java
BaseX 8.5.1 [Standalone]
Usage: basex [-bcdiIoqrRstuvVwxXz] [input]
  [input]     XQuery or command file, or query string
  -b<pars>    Bind external query variables
  -c<input>   Execute commands from file or string
  -d          Activate debugging mode
  -i<input>   Assign file or database to context
  -I<input>   Assign input string to context
  -o<output>  Write output to file
  -q<expr>    Execute XQuery expression
  -r<num>     Set number of query executions
  -R          Turn query execution on/off
  -s<pars>    Set serialization parameter(s)
  -t[path]    Run tests in file or directory
  -u          Write updates back to original files
  -v/V        Show (all) process info
  -w          Preserve whitespaces from input files
  -x          Show query plan
  -X          Show query plan before/after compilation
  -z          Skip output of results
```
In GUI mode, three are still missing, but instead of "tagsoup", it is now "jline":

```
mbwiesbaden@ubuntu-wiesbaden:~$ basexgui -h
[warning] /usr/bin/basexgui: Unable to locate /usr/share/java/jline.jar in /usr/share/java
[warning] /usr/bin/basexgui: Unable to locate /usr/share/java/xml-resolver.jar in /usr/share/java
[warning] /usr/bin/basexgui: Unable to locate /usr/share/java/jing.jar in /usr/share/java
BaseX 8.5.1 [GUI]
Usage: basexgui [-d] [file]
  [file]  Open specified XML or XQuery file
  -d  Activate debugging mode
```
I cannot tell why this happens and could not find help on the web, so thanks for your support.

**Comment by holu** [comment]
Hi Monika,

if I remember correctly then the mentioned libs are 'recommended' (but not mandatory) dependencies for BaseX.
Therefor only a 'warning' is emitted, since BaseX can be used without, but is missing some extras.
You can check that via, for instance, aptitude(1) or any other debian/ubuntu package manager. Recommended packages are listed for any package and can manually be chosen to be installed (as separate packages) along with a package.

An alternative would be to consider the .zip distribution from 

	http://files.basex.org/releases/9.3.3/

It has all dependencies included (and 9.3.3 would be a more recent version ;-)

All the best,
	Alex

**Comment by MonikaBarget** [comment]
Thanks, that was very helpful.
